From e4c733bae3459f5c29049b4f0e0c21e544a446c0 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Mon, 29 Jul 2019 15:55:01 +0530
Subject: [PATCH 0242/1921] octeontx2-pf: Always show physical link status

Irrespective of interface status being IFF_UP/DOWN,
always show physical interface status ie speed, duplex
etc via ethtool.

Change-Id: I570d66e76332071d28d600b57e0db8ea2ff58d3c
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/13435
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/13443
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../ethernet/marvell/octeontx2/nic/otx2_ethtool.c | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index fad6a0350a3e..72bf8541fc83 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -990,7 +990,6 @@ static int otx2_get_link_ksettings(struct net_device *netdev,
 				   struct ethtool_link_ksettings *cmd)
 {
 	struct otx2_nic *pfvf = netdev_priv(netdev);
-	bool if_up = netif_running(netdev);
 	u32 supported = 0, advertising = 0;
 	struct cgx_fw_data *rsp = NULL;
 
@@ -999,11 +998,6 @@ static int otx2_get_link_ksettings(struct net_device *netdev,
 	cmd->base.autoneg = pfvf->linfo.an;
 	cmd->base.port = pfvf->linfo.port;
 
-	if (!if_up) {
-		cmd->base.duplex = DUPLEX_UNKNOWN;
-		cmd->base.speed = SPEED_UNKNOWN;
-	}
-
 	rsp = otx2_get_fwdata(pfvf);
 	if (IS_ERR(rsp))
 		return PTR_ERR(rsp);
@@ -1256,13 +1250,8 @@ static int otx2vf_get_link_ksettings(struct net_device *netdev,
 
 	if (pfvf->pdev->device ==  PCI_DEVID_OCTEONTX2_RVU_AFVF) {
 		cmd->base.port = PORT_OTHER;
-		if (!netif_running(netdev)) {
-			cmd->base.duplex = DUPLEX_UNKNOWN;
-			cmd->base.speed = SPEED_UNKNOWN;
-		} else {
-			cmd->base.duplex = DUPLEX_FULL;
-			cmd->base.speed = SPEED_100000;
-		}
+		cmd->base.duplex = DUPLEX_FULL;
+		cmd->base.speed = SPEED_100000;
 	} else {
 		return	otx2_get_link_ksettings(netdev, cmd);
 	}
-- 
2.31.1

